skimage/filters/thresholding.py

```diff
@@ -335,15 +337,20 @@ def threshold_li(image):
     >>> thresh = threshold_li(image)
     >>> binary = image > thresh
     """
+    assert_nD(image, 2)
     # Requires positive image (because of log(mean))
-    offset = image.min()
-    # Can not use fixed tolerance for float image
-    imrange = image.max() - offset
-    image -= offset
+    offset = img.min()
+    if offset < 0:
+        img = image.copy()
+        img -= offset
+    else:
+        img = image  # avoid a copy
+
+    imrange = img.max()
+    tolerance = 0.5 * imrange / 256.
```

Comment:
(I think you can drop it even without that, since the `0.5 *` will resolve first.